src/main/java/com/mmaarten/mmcommands/MmCommandHandlerBuilder.java

```diff
@@ -0,0 +1,401 @@
+package com.mmaarten.mmcommands;
+
+import org.bukkit.ChatColor;
+import org.bukkit.command.CommandSender;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * <p>
+ * Use this builder to construct a {@link MmCommandHandler}
+ * instance when
+ * you need to set configuration options other than the default. For an
+ * {@link MmCommandHandler} instance with default configuration, you
+ * can simply use {@code new MmCommandHandler()}.
+ * </p>
+ *
+ * <p>
+ * The following shows an example of how to use the
+ * {@link MmCommandHandlerBuilder} to construct an
+ * {@link MmCommandHandler} instance:
+ * </p>
+ *
+ * <pre>
+ * MmCommandHandler mmCommandHandler = new MmCommandHandlerBuilder()
+ *     .enableHelpSubCommand()
+ *     .generateTabCompletions()
+ *     .runLastAllowed()
+ *     .setHelpHeader("--------------Help---------------")
+ *     .setAmountOfCommandsPerHelpPage(15)
+ *     .build();
+ * </pre>
+ *
+ * NOTE: the order of invocation of the configuration methods does not
+ * matter.
+ * @author Mmaarten
+ * @author TheSummerGrinch
+ */
+public final class MmCommandHandlerBuilder {
+
+    protected final @NotNull List<MmCommand> commands = new ArrayList<>();
+    protected int commandsPerPage = 5;
+    protected boolean useHelp = false;
+    protected boolean generateTabCompletions = false;
+    protected boolean runLastAllowed = false;
+    @NotNull protected String helpHeader = "----Help---- page: %page%";
+    @NotNull protected String helpCommandPrefix = "&d";
+    @NotNull protected String helpCommandArgumentSpacer = " &5";
+    @NotNull protected String helpArgumentDescriptionSpacer = " &8> &7";
+    @NotNull protected String helpPropertyPrefix = "&5";
+    @NotNull protected String helpPropertyValueSpacer = " &8> &7";
+    private boolean commandHandlerBuilt = false;
+
+    /**
+     * Creates an MmCommandBuilder instance, preloaded with the default
+     * configuration, that can be used to build a {@link MmCommandHandler}
+     * instance with a custom configuration. MmCommandHandlerBuilder is
+     * typically used by first invoking various configuration methods to set
+     * desired options, and finally calling {@link #build()}.
+     */
+    public MmCommandHandlerBuilder() {}
+
+    /**
+     * Constructs a MmCommandHandlerBuilder instance from an existing
+     * MmCommandHandler. The newly constructed MmCommandHandlerBuilder will
+     * be created with the same configuration as the existing
+     * MmCommandHandler instance.
+     * @param commandHandler the instance of which the configuration will be
+     *                       applied to the new MmCommandHandlerBuilder.
+     */
+    MmCommandHandlerBuilder(MmCommandHandler commandHandler) {
```

Comment:
Should be a public constructor and seeing that there is no way to remove commands from a MmCommandHandler, the commands should not be copied. Just the configuration